solar_system.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'planet'
+
+# Building a SolarSystem class
+# which is responsible for keeping track of a collection of instances of Planet.
+class SolarSystem
+
+# star_name parameter is saved as an instance variable.
+# also an instance variable called @planets, to store an array of planets.
+# @planets is set as an empty array.
+# Both @star_name and @planets are readable but not writable.
+
+  attr_reader :star_name,:planets
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+# New method which will take an instance of Planet
+# as a parameter and add it to the list of planets.
+  def add_planet(planet)
+    @planets << planet
+    return @planets
+  end
+
+ # New method which will return
+ # a string containing a list of all the planets
+ # in the system.
+  def list_planets
+    planet_list = "Planets orbiting #{@star_name} "
+    i = 1
+    @planets.each do |planet|
+      planet_list += "\n #{i}. #{planet.name}"
+      i += 1
+    end
+    return planet_list
+  end
+
+# New method that takes the name of a planet as a string,
+# and returns the corresponding instance of Planet.
+# The lookup should be case-insensitive, so that Earth, earth and eArTh all return the same planet.
+
+  def find_planet_by_name(planet_string)
+    @planets.each do |planet|
```

Comment:
A good start.  It **doesn't work** because the 2nd return statement inside the `.each` loop.  If the 1st planet doesn't match, it returns "This planet doesn't exist"

You could fix it by moving that 2nd return to outside the loop.


You could also do:

```ruby
planet_string = planet_string.downcase
return planets.find do |planet|
  planet.name.downcase == planet_string
end
```

In the above it would return `nil` if it could not find the planet.